src/isomorphic/devtools/ReactInvalidSetStateWarningDevTool.js

```diff
@@ -0,0 +1,32 @@
+
+'use strict';
+
+var warning = require('warning');
+
+if (__DEV__) {
+  var processingChildContext = false;
+
+  var warnInvalidSetState = function() {
+    if (processingChildContext === false) {
+      return;
+    }
+    warning(
```

Comment:
It might be more compact to write

```js
warning(
  !processingChildContext,
  'setState(...): Cannot call setState inside getChildContext()'
);
```